core/blockchain.go

```diff
@@ -2394,6 +2394,16 @@ func (bc *BlockChain) maintainTxIndex() {
 	}
 	defer sub.Unsubscribe()
 
+	// Launch the initial processing if chain is not empty. This step is
+	// useful in these scenarios that chain has no progress and indexer
+	// is never triggered.
+	head := rawdb.ReadHeadBlock(bc.db)
+	tail := rawdb.ReadTxIndexTail(bc.db)
+	if head != nil && tail != nil {
+		done = make(chan struct{})
+		bc.indexBlocks(tail, head.NumberU64(), done)
```

Comment:
missing `go` here

Reply:
As discussed with @rjl493456442 we should run it in sync mode.

Reply:
I don't know about this discussion. But if we change it to sync here, it means the process might take a long time to exit when geth is aborted while initial tx indexing is running?

Reply:
> it means the process might take a long time to exit when geth is aborted while initial tx indexing is running?

I don't think this will happen, as `bc.quit` is directly passed into the `rawdb.IndexTransactions`, `rawdb.UnindexTransactions`, so if `bc.quit` is closed, those processes will fast quit.